aries_cloudagent/connections/models/diddoc/schemas/diddocdeserializeschema.py

```diff
@@ -0,0 +1,14 @@
+from marshmallow import Schema, fields
+from .publickeyschema import PublicKeySchema
+from .serviceschema import ServiceSchema
+from .authenticationschema import AuthenticationSchema
+from .verificationmethodschema import VerificationMethodSchema
+
+class DIDDocSchema(Schema):
+
+    id = fields.Str()
+    publicKey = fields.List(fields.Nested(PublicKeySchema))
+    authentication = fields.List(fields.Nested(AuthenticationSchema))
+    service = fields.List(fields.Nested(ServiceSchema))
+    verificationMethod = fields.Nested(VerificationMethodSchema)
```

Comment:
Values in the verification method list are not homogeneous. I believe this applies to other DID core values as well. In the list you can have either a Verification method map OR a DID URL reference to another DID Doc value.

Reply:
With the following DIDDoc as example:

> {
   "@context":[
      "https://www.w3.org/ns/did/v1"
   ],
   "id":"did:sov:CYQLsccvwhMTowprMjGjQ6",
   "verificationMethod":[
      {
         "type":[
            "Ed25519VerificationKey2018"
         ],
         "id":"did:sov:CYQLsccvwhMTowprMjGjQ6#key-1",
         "publicKeyBase58":"CLFRfp2wa3ifbsVvdq52WcpEy7aujactsoqQgxkz7ZKR"
      }
   ],
   "authentication":[
      {
         "type":[
            "Ed25519SignatureAuthentication2018"
         ],
         "verificationMethod":"did:sov:CYQLsccvwhMTowprMjGjQ6#key-1"
      }
   ],
   "service":[
      {
         "type":[
            "custom"
         ],
         "serviceEndpoint":"https://notary.ownyourdata.eu/?hash=fef73e27304f4f9c17655f1c598b75237664aff67e48a50edc51505a742985cf"
      }
   ]
}

...

Looking deeper in the verificationMethod:

> verificationMethod":[

      {
         "type":[
            "Ed25519VerificationKey2018"
         ],
         "id":"did:sov:CYQLsccvwhMTowprMjGjQ6#key-1",
         "publicKeyBase58":"CLFRfp2wa3ifbsVvdq52WcpEy7aujactsoqQgxkz7ZKR"
      }
   ]

It is the same from a logic perspective the verificationMethod as a dict or a list?
The type parameter must be a string or it is also posible to be "readed" as a list?

As far I has understood, if we serialize the metadata we will loose the json structure, but that does not matter because we have "the recipe" to compose the same JSON. 

If I assume that the parameter "verificationMethod" must be always a list of at least one element, and the same for the type, It would have impact in the project?

Reply:
From the DID Spec:
> The value of the type property MUST be exactly one verification method type expressed as a string.

So the type isn't a list. What I was referring to was that the verification method list may look like:

```
{
  ...
  "verificationMethod": [
    { <regular verification method attributes> },
    "did:sov:1234abcd#key-1"
  ],
}
```

Reply:
Oops, and I'm actually wrong there. See this table for where I'm seeing my mistake now:
https://w3c.github.io/did-core/#did-document-properties

I'm not sure if that was changed recently or if I was simply mistaken, my aplogies.

Reply:
```
"verificationMethod": [{
    "id": ...,
    "type": ...,
    "controller": ...,
    "publicKeyJwk": ...
  }, {
    "id": ...,
    "type": ...,
    "controller": ...,
    "publicKeyBase58": ...
  }]
```

Reply:
### Verification Method Properties from DID Spec
Property | Required? | Value constraints
------------|---------------|------------------------
id	|yes|	A string that conforms to the rules in Section § 3.2 DID URL Syntax.
controller	|yes|	A string or an ordered set of strings that conform to the rules in Section § 3.1 DID Syntax.
type	|yes|	A string.
publicKeyJwk	|no|	A map representing a JSON Web Key that conforms to [RFC7517]. See definition of publicKeyJwk for additional constraints.
publicKeyBase58	|no|	A string that conforms to a base58btc encoded public key.